File: app/src/main/java/com/ichizin/hatezin/repository/HatenaClient.java
package com.ichizin.hatezin.repository;

import com.ichizin.hatezin.model.HatenaFeed;

import javax.inject.Inject;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.simplexml.SimpleXmlConverterFactory;
import retrofit2.http.GET;
import rx.Observable;

/**
 *
 *
 * @author ichizin
 */
public class HatenaClient {

    private static final String BASE_URL = "http://b.hatena.ne.jp";
    HatebuService hatebuService;

    @Inject
    HatenaClient(OkHttpClient okHttpClient) {

        Retrofit hatebuRetrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(okHttpClient)
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .addConverterFactory(SimpleXmlConverterFactory.create())
                .build();

        hatebuService = hatebuRetrofit.create(HatebuService.class);
    }

    public Observable<HatenaFeed> getHotEntry() {
        return this.hatebuService.getHotEntry();
    }

    public Observable<HatenaFeed> getSocial() {
        return this.hatebuService.getSocial();
    }

    public Observable<HatenaFeed> getEconomics() {
        return this.hatebuService.getEconomy();
    }

    public Observable<HatenaFeed> getLife() {
        return this.hatebuService.getLife();
    }

    public Observable<HatenaFeed> getKnowledge() {
        return this.hatebuService.getKnowledge();
    }

    public Observable<HatenaFeed> getIt() {
        return this.hatebuService.getIt();
    }

    public Observable<HatenaFeed> getEntertaiment() {
        return this.hatebuService.getEntertainment();
    }

    public Observable<HatenaFeed> getGame() {
        return this.hatebuService.getGame();
    }

    public Observable<HatenaFeed> getFun() {
        return this.hatebuService.getFun();
    }

    public Observable<HatenaFeed>getVideo() {
        return this.hatebuService.getVideo();
    }


    interface HatebuService {

        @GET("/hotentry.rss")
        Observable<HatenaFeed> getHotEntry();

        @GET("/hotentry/social.rss")
        Observable<HatenaFeed> getSocial();

        @GET("/hotentry/economics.rss")
        Observable<HatenaFeed> getEconomy();

        @GET("/hotentry/life.rss")
        Observable<HatenaFeed> getLife();

        @GET("/hotentry/knowledge.rss")
        Observable<HatenaFeed> getKnowledge();

        @GET("/hotentry/it.rss")
        Observable<HatenaFeed> getIt();

        @GET("/hotentry/entertainment.rss")
        Observable<HatenaFeed> getEntertainment();

        @GET("/hotentry/game.rss")
        Observable<HatenaFeed> getGame();

        @GET("/hotentry/fun.rss")
        Observable<HatenaFeed> getFun();

        @GET("/video.rss")
        Observable<HatenaFeed> getVideo();

    }

}

File: app/src/main/java/com/ichizin/hatezin/repository/HatenaClient.kt
package com.ichizin.hatezin.repository

import javax.inject.Inject

import okhttp3.OkHttpClient
import retrofit.Retrofit
import retrofit.RxJavaCallAdapterFactory
import retrofit2.converter.simplexml.SimpleXmlConverterFactory

/**
 * HatenaClient

 * @author ichizin
 */
class HatenaClient {

    val HOT_ENTRY_URL = "http://b.hatena.ne.jp/hotentry.rss"
    val SOCIAL_URL = "http://b.hatena.ne.jp/hotentry/social.rss"
    val ECONOMICS_URL = "http://b.hatena.ne.jp/hotentry/economics.rss"
    val LIFE_URL = "http://b.hatena.ne.jp/hotentry/life.rss"
    val KNOWLEDGE_URL = "http://b.hatena.ne.jp/hotentry/knowledge.rss"
    val IT_URL = "http://b.hatena.ne.jp/hotentry/it.rss"
    val ENTER_TAINMENT_URL = "http://b.hatena.ne.jp/hotentry/entertainment.rss"
    val GAME_URL = "http://b.hatena.ne.jp/hotentry/game.rss"
    val FUN_URL = "http://b.hatena.ne.jp/hotentry/fun.rss"
    val VIDEO_URL = "http://b.hatena.ne.jp/video.rss"

    val BASE_URL = "http://b.hatena.ne.jp/"

    @Inject
    constructor(okHttpClient: OkHttpClient) {

//        var retrofit: Retrofit = Retrofit.Builder()
//                .client(okHttpClient)
//                .baseUrl(BASE_URL)
//                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
//                .addConverterFactory(SimpleXmlConverterFactory.create())
//                .build()
    }

    interface HatebuServide {


    }
}

